sympy/solvers/solveset.py

```diff
@@ -978,6 +985,88 @@ def _solveset(f, symbol, domain, _check=False):
     return result
 
 
+def _expo_solver(f, symbol):
+    """
+    Helper function for solving exponential equations.
+
+    This function solves exponential equation by using logarithms.
+    Exponents are converted to a more general form of logs which can further be
+    solved by passing to _solveset.
+    It deals with equations of type `a*f(x) + b*g(x)`, where f(x) and g(x)
+    are power terms.
+
+    eg: 3**(2*x) - 2**(x + 3) can be transformed to a better log form,
+    2*x*log(3) - (x+3)*log(2), which is easily solvable.
+    """
+
+    a, b = ordered(f.args)
+
+    lhs = a
+    rhs = -b
+
+    lhs = expand_log(log(lhs), force=True)
+    rhs = expand_log(log(rhs), force=True)
```

Comment:
force=True does operations that are not mathematically correct for all values. Is it possible for this function to return wrong results because of this?

Reply:
Yes, log(x*y) will expand into log(x)+log(y) but that is not always true. e.g. if x == y == -1, the former is 0 while the latter is `2*pi*I` in SymPy.